mypy/nodes.py

```diff
@@ -512,7 +506,9 @@ class FuncBase(Node):
         "is_static",  # Uses "@staticmethod"
         "is_final",  # Uses "@final"
         "_fullname",
-        "is_dataclass_transform",  # Is decorated with "@typing.dataclass_transform" or similar
+        # Present when a function is decorated with "@typing.dataclass_transform" or similar, and
+        # records the parameters passed to typing.dataclass_transform for later use
+        "dataclass_transform_spec",
```

Comment:
Do we need to add this to `FuncBase`? Could we only have this only in `FuncDef` nodes? Overloaded functions could then extract it from the items if needed, perhaps.

Reply:
My idea would be to store the spec only in `FuncDef` (i.e. have an attribute for it), and have a helper method in `OverloadedFuncDef` to find the spec from the overload items or the implementation on demand. This might be an abstract method in `FuncBase` that `FuncDef` could also implement and directly return the spec (not sure if this would be needed though). I'd prefer to avoid having the spec stored in multiple places, as these could get out of sync or just generally add confusion -- but it's not a big deal either way.

If this doesn't make sense, it's also fine to store it in `FuncBase`.

Reply:
That's reasonable; even if it doesn't pan out, I can revisit that when I fully implement overloads